### PR TITLE
Add `Actions` feature toggle to the database settings page

### LINKED ISSUE #22545: Add a toggle to enable/disable writeback to the database page
Part of #22542

Add a toggle or other sort of UI control to the admin database page (`/admin/database/:id`) to control if writeback is enabled for the database. The control should only be displayed if the "writeback" global feature flag is turned on.

### PR DESCRIPTION
**Pull request by akiselev**
Resolves #22545 
Part of #22542

Adds an option to enable experimental actions

image

**NOTE** This depends on #22691 and #22689 to work

### CODE REVIEW

File: src/metabase/driver/postgres.clj
Comment (akiselev) [inline src/metabase/driver/postgres.clj:172]: @camsaul is there a correct way to add a dependency on global settings on the backend or this is a frontend only value?

File: src/metabase/driver/postgres.clj
Comment (akiselev) [inline src/metabase/driver/postgres.clj:170]: @camsaul I used `deferred-tru` because that's what `common.clj` uses for settings but the rest of the code here uses `trs`. Is one preferred over the other? Also, is this setting name supposed to mirror the `defsetting` in #22559 ?

@camsaul @kulyk is this the language we want to use on the database settings page?

Comment (camsaul) [inline src/metabase/driver/postgres.clj:170]: `database-enable-actions` isn't something that goes in Database `details`... it goes in Database `settings`. So you don't need to add this.

Comment (camsaul) [reply src/metabase/driver/postgres.clj:170]: I need to implement https://github.com/metabase/metabase/issues/19409 really quick so you can edit the Database-local Setting value

File: src/metabase/driver/postgres.clj
Comment (camsaul) [reply src/metabase/driver/postgres.clj:172]: Since this is a Database-local Setting and not something that goes in the details map you should probably just put all this info in the FE somewhere

**Comment by camsaul** [comment]
Just to clarify `database-enable-actions` is a [Database-local Setting](https://github.com/metabase/metabase/pull/19399) which means it lives in the Database `settings` column, *not* `details`. `details` is for connection properties and stuff like that. `settings` is for Settings. The difference is `details` is considered sensitive and changes to it mean we need to flush connection pools. 

Other stuff like the Database `is_full_sync`, `auto_run_queries`, `refingerprint`, and the sync schedule columns are all things that are conceptually similar, so you should do the UI for this the same way you do for those columns -- it should be defined in the FE and then rather than in the driver like we do for connection properties. (Sync schedule and those other flags will probably get moved into `settings` at some point -- it was overkill to have them be separate columns in the first place.)

I need to implement #19409 really quick so you can fetch and set Database `settings`. I will get that done today

File: src/metabase/driver/postgres.clj
Comment (dpsutton) [reply src/metabase/driver/postgres.clj:170]: I put the persistence enabled information into the options ([pg driver](https://github.com/metabase/metabase/blob/master/src/metabase/driver/postgres.clj#L54-L56)) but i suppose settings makes more sense. We should switch that over.

Comment (kulyk) [reply src/metabase/driver/postgres.clj:170]: > is this the language we want to use on the database settings page?

I think this one works at the moment 👍

**Comment by camsaul** [comment]
You can already get the value of Database-local Settings with `GET /api/database/` or `GET /api/database/:id` -- they come back under the `settings` key. 

https://github.com/metabase/metabase/pull/22689 adds support for settings the values via `PUT /api/database/:id`

**Comment by camsaul** [comment]
RE only showing this for specific databases: you shouldn't hardcode this to Postgres specifically since there's no real reason a lot of this stuff needs to be Postgres-only. Instead you should check the `experimental-enable-actions` Setting (which is a global feature flag).

Also once I implement #22557 you should check that the driver supports the `actions` feature or whatever I end up calling it. I'd say just checking the global feature flag is ok for right now tho. Worst-case scenario you mark it enabled for a Database and it just doesn't do anything

**Comment by camsaul** [comment]
PR https://github.com/metabase/metabase/pull/22691 adds the `:actions` feature flag so once that's merged you can check that

**Comment by akiselev** [comment]
@camsaul I reimplemented it using the database settings, can you give it another spin?

**Review by kulyk — APPROVED** [review]
Looks good to me! Let's wait for Cam's response though

File: frontend/src/metabase-types/types/Database.ts
Comment (camsaul) [reply frontend/src/metabase-types/types/Database.ts:26]: I think that makes sense

**Review by camsaul — APPROVED** [review]
This looks good to me but once #22691 is merged in we'll probably _also_ want to add a check to make sure the Database supports the `:actions` feature. After all even if the global `experimental-enable-actions` Setting is enabled you still shouldn't be able to enable it for MongoDB for example since we're not support that just yet.

**Pull request merged**
